### PR TITLE
Add get_connectivity_state to Channel

### PR DESCRIPTION
**Pull request by zach-youssef**
Exposes the channel connectivity state struct and getter method from grpc::sys to the channel implementation.

### CODE REVIEW

File: src/channel.rs [hunk 1]
```diff
@@ -562,6 +573,10 @@ impl Channel {
         }
     }
 
+    pub fn check_connectivity_state(&self, try_to_connect: bool) -> ConnectivityState {
```
Comment (BusyJay) [inline src/channel.rs]: Better add some comment to explain the side effect of the function.

File: src/channel.rs [hunk 2]
```diff
@@ -531,6 +533,15 @@ struct ChannelInner {
     channel: *mut GrpcChannel,
 }
 
+impl ChannelInner{
+    pub(crate) fn check_connectivity_state(&self, try_to_connect: bool) -> ConnectivityState {
```
Comment (BusyJay) [inline src/channel.rs]: It's OK to keep the implementation in the function at L576.

**Comment by zach-youssef** [comment]
Would you prefer for me to squash my fix commit or leave as-is?

**Comment by Hoverbear** [comment]
@zach-youssef When the PR is merged we'll squash. Adding them as-is is convenient for code review.

**Comment by Hoverbear** [comment]
@zach-youssef A `cargo fmt` is required to get the build to pass. :(

**Comment by Hoverbear** [comment]
@BusyJay I resolved it, PTAL.

**Pull request merged**